### DISCUSSION TITLE
Airflow dynamic task mapping is not working with Datahub's Airflow lineage plugin

### DISCUSSION THREAD

**Discussion by langelrhea** [apache/airflow#26129] [General]
### Apache Airflow version

2.3.4

### What happened

Working with Airflow if you follow the steps from Datahub Integration (https://datahubproject.io/docs/lineage/airflow/) the first step is install the required dependency in your airflow (pip install acryl-datahub-airflow-plugin) after install this dependency, if I run my DAG this one fails. However if you uninstall this package, this DAG is completed successfully. 

### How to reproduce

Execute a DAG and see that this one is completed, after that install datahub plugin (pip install acryl-datahub-airflow-plugin) this DAG won't be completed, then if you uninstall the same package (pip uninstall acryl-datahub-airflow-plugin) this DAG will be completed as the first time.

### Operating System

Ubuntu 20.04.5 LTS (Focal Fossa)

### Versions of Apache Airflow Providers

apache-airflow-providers-cncf-kubernetes == 4.3.0
apache-airflow-providers-ftp==3.0.0
apache-airflow-providers-http==3.0.0
apache-airflow-providers-imap==3.0.0
apache-airflow-providers-sqlite==3.0.0

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by uranusjr** [comment]
Have you contacted the plugin's maintainers? This sounds like a problem with the plugin, and even if not, its maintainers should have a better understanding on what is causing the issue.

If you are convinced that this is caused by something in the main Airflow codebase, some logs to provide insight on what's going on would be very helpful. There is close to nothing we can do with the little information available here.

**Comment by potiuk** [comment]
Surely this is an interesting issue and good point but totally mis-directed. You should raise it in DataHub as they are suporting it not us.

**Comment by potiuk** [comment]
I converted it to a discussion - and if you get more information after talking to datahub, we can continue discussing here @langelrhea

**Comment by langelrhea** [comment]
Thank you so much for your comment! Sorry for the misunderstanding, I will keep this discussion updated in order to help similar issues from others users

**Reply by potiuk** [reply]
Surely. It would be fantastic if you come back after talking to DataHub and report your findings here.
